server/src/main/java/org/apache/druid/server/coordinator/duty/NewestSegmentFirstIterator.java

```diff
@@ -247,17 +248,30 @@ public boolean hasNext()
     }
   }
 
-  private boolean needsCompaction(DataSourceCompactionConfig config, SegmentsToCompact candidates)
+  @VisibleForTesting
+  static PartitionsSpec findPartitinosSpecFromConfig(ClientCompactionTaskQueryTuningConfig tuningConfig)
+  {
```

Comment:
The auto compaction supported only dynamic partitioning before this PR. The deprecated `maxRowsPerSegment` and `maxTotalRows` in `DataSourceCompactionConfig` are for dynamic partitioning. Users are recommended to use `partitionsSpec` instead of them. 

> Misread the code. if the maxTotalRows is null in partition spec, should it be read from tuningConfig instead of using the Long.Max?

My intention is using `partitionsSpec` if it's specified and ignoring deprecated ones. I will add docs for this behaviour as well as the newly supported tuning configurations.